simulationdataschema/atoms_state.py

```diff
@@ -0,0 +1,571 @@
+    )
+
+    u_effective = Quantity(
+        type=np.float64,
+        unit="joule",
+        description="""
+        Value of the effective U parameter (u - j).
+        """,
+    )
+
+    slater_integrals = Quantity(
+        type=np.float64,
+        shape=[3],
+        unit="joule",
+        description="""
+        Value of the Slater integrals [F0, F2, F4] in spherical harmonics used to derive
+        the local Hubbard interactions:
+
+            u = ((2.0 / 7.0) ** 2) * (F0 + 5.0 * F2 + 9.0 * F4) / (4.0*np.pi)
+
+            up = ((2.0 / 7.0) ** 2) * (F0 - 5.0 * F2 + 3.0 * 0.5 * F4) / (4.0*np.pi)
+
+            jh = ((2.0 / 7.0) ** 2) * (5.0 * F2 + 15.0 * 0.25 * F4) / (4.0*np.pi)
+
+        See e.g., Elbio Dagotto, Nanoscale Phase Separation and Colossal Magnetoresistance,
+        Chapter 4, Springer Berlin (2003).
+        """,
+    )
+
+    double_counting_correction = Quantity(
+        type=str,
+        description="""
+        Name of the double counting correction algorithm applied.
+        """,
+        a_eln=ELNAnnotation(component="StringEditQuantity"),
+    )
+
+    def resolve_u_interactions(self, logger: BoundLogger) -> Optional[tuple]:
+        """
+        Resolves the Hubbard interactions (u, up, jh) from the Slater integrals (F0, F2, F4).
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Optional[tuple]): The Hubbard interactions (u, up, jh).
+        """
+        if self.slater_integrals is None or len(self.slater_integrals) == 3:
+            logger.warning(
+                "Could not find `slater_integrals` or the length is not three."
+            )
+            return None
+        f0 = self.slater_integrals[0]
+        f2 = self.slater_integrals[1]
+        f4 = self.slater_integrals[2]
+        u_interaction = (
+            ((2.0 / 7.0) ** 2)
+            * (f0 + 5.0 * f2 + 9.0 * f4)
+            / (4.0 * np.pi)
+            * ureg("joule")
+        )
+        up_interaction = (
+            ((2.0 / 7.0) ** 2)
+            * (f0 - 5.0 * f2 + 3.0 * f4 / 2.0)
+            / (4.0 * np.pi)
+            * ureg("joule")
+        )
+        jh_interaction = (
+            ((2.0 / 7.0) ** 2)
+            * (5.0 * f2 + 15.0 * f4 / 4.0)
+            / (4.0 * np.pi)
+            * ureg("joule")
+        )
+        return u_interaction, up_interaction, jh_interaction
+
+    def resolve_u_effective(self, logger: BoundLogger) -> Optional[np.float64]:
+        """
+        Resolves the effective U parameter (u - j).
+
+        Args:
+            logger (BoundLogger): The logger to log messages.
+
+        Returns:
+            (Optional[np.float64]): The effective U parameter.
+        """
+        if self.u is None or self.j is None:
+            logger.warning(
+                "Could not find `HubbardInteractions.u` or `HubbardInteractions.j`."
+            )
+            return None
+        return self.u - self.j
+
```

Comment:
Nice! Do we have unit tests for these?

Reply:
Not yet, maybe we can tackle testing soon in another branch. For now I am testing directly on a parser and checking that everything goes well.

Reply:
Do you mind opening this as an issue? Maybe you can also share your ideas of what you are expecting for testing and we discuss there.

Reply:
I can also take it after you merge this and add you as a reviewer?
I'll open the issue this evening. Just remind me if I forget.